ocsp_status.go

```diff
@@ -39,33 +48,47 @@ func checkFromCert(tools ocsptools.ToolsInterface, linter linter.LinterInterface
 		return fmt.Errorf("Error parsing certificate from certificate file: %w", err)
 	}
 
+	var issuerCert *x509.Certificate
 	h := helpers.Helpers{}
 
-	issuerCert, err := tools.GetIssuerCertFromLeafCert(h, leafCert)
-	if err != nil {
-		return fmt.Errorf("Error getting issuer certificate from certificate: %w", err)
+	if issuerFile != "" {
+		issuerCert, err = tools.ParseCertificateFile(issuerFile)
+		if err != nil {
+			return fmt.Errorf("Error parsing certificate from certificate file: %w", err)
+		}
+	} else {
+		issuerCert, err = tools.GetIssuerCertFromLeafCert(h, leafCert)
+		if err != nil {
+			return fmt.Errorf("Error getting issuer certificate from certificate: %w", err)
+		}
 	}
 
 	ocspResp, err := tools.FetchOCSPResp(h, ocspURL, dir, leafCert, issuerCert, reqMethod, hash)
 	if err != nil {
 		return fmt.Errorf("Error fetching OCSP response: %w", err)
 	}
 
-	linter.LintOCSPResp(ocspResp)
+	linter.LintOCSPResp(ocspResp, leafCert, verbose)
 
 	return nil
 }
 
 // checkFromURL takes a server URL and constructs and sends an OCSP request to
 // check that URL's certificate then parses and lints the OCSP response
-func checkFromURL(tools ocsptools.ToolsInterface, linter linter.LinterInterface, serverURL string, shouldPrint bool, isPost bool, noStaple bool, ocspURL string, dir string, hash crypto.Hash) error {
+func checkFromURL(tools ocsptools.ToolsInterface, linter linter.LinterInterface, serverURL string, issuerFile string, shouldPrint bool, isPost bool, noStaple bool, ocspURL string, dir string, hash crypto.Hash, verbose bool) error {
```

Comment:
Similar to my other comment, I would think that all this does it get the cert from the URL and call checkFromCert so that everything has a common entry point and there is less duplicate code.

Reply:
So `checkFromCert` reads in a certificate file, whereas `checkFromURL` gets the certificate chain from establishing a TLS connection with the URL. There is definitely overlapping functionality, which I have done my best to refactor, but I think it would be awkward to force one to call the other.